From 4a437024d3a8e8384251b126b3d97cb7e7cda4ba Mon Sep 17 00:00:00 2001
From: Zachary Waldowski <zwaldowski@gmail.com>
Date: Fri, 27 Apr 2012 11:04:00 -0400
Subject: [PATCH 4/5] Some modern unified-syntax assemblers don't like
 macros-with-arguments; we don't use it anyway for RETLDM.

---
 src/arm/sysv.S | 16 ++++------------
 1 file changed, 4 insertions(+), 12 deletions(-)

diff --git a/src/arm/sysv.S b/src/arm/sysv.S
index 62cffaa..30a490f 100644
--- a/src/arm/sysv.S
+++ b/src/arm/sysv.S
@@ -148,20 +148,12 @@ _L__\name:
 .endm
 #endif
 
-.macro	RETLDM	regs=, cond=, dirn=ia
+.macro	RETLDM
 #if defined (__INTERWORKING__)
-	.ifc "\regs",""
-	ldr\cond	lr, [sp], #4
-	.else
-	ldm\cond\dirn	sp!, {\regs, lr}
-	.endif
-	bx\cond	lr
+	ldr	lr, [sp], #4
+	bx	lr
 #else
-	.ifc "\regs",""
-	ldr\cond	pc, [sp], #4
-	.else
-	ldm\cond\dirn	sp!, {\regs, pc}
-	.endif
+	ldr	pc, [sp], #4
 #endif
 .endm
 
-- 
1.8.4

